demo/vl-form-validation.html

```diff
@@ -99,8 +98,8 @@
                         <label is="vl-form-label" for="select-1" data-vl-block>Stad</label>
                     </div>
                     <div is="vl-form-column" data-vl-size="9">
-                        <select id="select-1" is="vl-select-demo" data-vl-block name="validation-select" 
-                            data-vl-required="true" 
+                        <select id="select-1" is="vl-select-demo" data-vl-block name="validation-select"
```

Comment:
ik zou hier weer select-1 nen betere naam geven?

Reply:
Kan ik aanpassen, hier is er echter enkel automatische formatting toegepast precies, overal aanpassen dan?

Reply:
ik vind select-1 nogal nutteloos qua naam maar als het veel werk is hoeft het nu ook per se niet. maar is maar wat namen wijzigen zeker

Reply:
Ik ben meestal ook voorstander van logische id's, maar we moeten bij de webcomponenten opletten dat we hier niet in overdrijven en daardoor de leesbaarheid net slechter wordt.

``` JS
it('als gebruiker zie ik een foutmelding als er niets is geselecteerd uit een lijst', async () => {
  const form = await vlFormValidationPage.getForm(1);
  const select = await form.getSelect(1);

  await form.submit();
  await assert.eventually.isTrue(select.hasError());

  await select.selectByIndex(1);
  await form.submit();
  await assert.eventually.isFalse(select.hasError());
});
```

De essentie in deze test is "als gebruiker zie ik een foutmelding als er niets is geselecteerd uit een lijst". Dat er toevallig steden in de select staan is in kader van deze test irrelevant.

De wijzigingen zoals hieronder zijn voor mij niet echt een meerwaarde, integendeel.

``` JS
it('als gebruiker zie ik een foutmelding als er niets is geselecteerd uit een lijst', async () => {
  const form = await vlFormValidationPage.getFormulierMetSteden();
  const select = await form.getStedenSelect();

  await form.submit();
  await assert.eventually.isTrue(select.hasError());

  await select.selectByIndex(1);
  await form.submit();
  await assert.eventually.isFalse(select.hasError());
});
```

Bij vl-button gebruiken we wel logische id's, omdat daar het id wel relevant is:
``` JS
it('als gebruiker wil ik het verschil kunnen zien tussen een secondary en een gewone knop', async () => {
  const primaryButton = await vlButtonPage.getPrimaryButton();
  const secondaryButton = await vlButtonPage.getSecondaryButton();
  await assert.eventually.isFalse(primaryButton.isSecondary());
  await assert.eventually.isTrue(secondaryButton.isSecondary());
});
```

Nu, ik kan me gerust ook neerleggen bij logische id's, maar lijkt me hier echt geen meerwaarde dus zou ik er geen tijd in steken.

Reply:
Is inderdaad niet veel werk en heb het maar aangepast